lib/omniauth/strategies/slack.rb

```diff
@@ -19,56 +19,37 @@ class Slack < OmniAuth::Strategies::OAuth2
         param_name: 'token'
       }
 
-      uid { raw_info['user_id'] }
+      # User ID is not guaranteed to be globally unique across all Slack users.
+      # The combination of user ID and team ID, on the other hand, is guaranteed
+      # to be globally unique.
+      uid { "#{user_identity['id']}-#{team_identity['id']}" }
 
       info do
         hash = {
-          nickname: raw_info['user'],
-          team: raw_info['team'],
-          user: raw_info['user'],
-          team_id: raw_info['team_id'],
-          user_id: raw_info['user_id']
+          name: user_identity['name'],
+          email: user_identity['email'],    # Requires the identity.email scope
+          image: user_identity['image_48'], # Requires the identity.avatar scope
```

Comment:
Is there are reason to pick the size `48`? Having a default is great tho, but what about adding `images` with all the sizes as keys?